nautobot/extras/jobs.py

```diff
@@ -104,25 +105,190 @@ class Meta:
+            raise RunJobTaskFailed(f"Unable to serialize data for job {task_id}") from err
+        except Exception as err:
+            raise RunJobTaskFailed(f"Unexpected failure in job {self.name}") from err
+
+        job_model = self.get_job_model()
+        if not job_model.enabled:
+            self.log_failure(
+                message=f"Job {job_model} is not enabled to be run!",
+                obj=job_model,
+            )
+
+        soft_time_limit = job_model.soft_time_limit or settings.CELERY_TASK_SOFT_TIME_LIMIT
+        time_limit = job_model.time_limit or settings.CELERY_TASK_TIME_LIMIT
+        if time_limit <= soft_time_limit:
+            self.log_warning(
+                f"The hard time limit of {time_limit} seconds is less than "
+                f"or equal to the soft time limit of {soft_time_limit} seconds. "
+                f"This job will fail silently after {time_limit} seconds.",
+            )
+
+        self.log_info("Running job")
+
+    def run(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        """
+        Method invoked when this Job is run.
+        """
+        raise NotImplementedError("Jobs must define the run method.")
+
+    def on_success(self, retval, task_id, args, kwargs):
+        """Success handler.
+
+        Run by the worker if the task executes successfully.
+
+        Arguments:
+            retval (Any): The return value of the task.
+            task_id (str): Unique id of the executed task.
+            args (Tuple): Original arguments for the executed task.
+            kwargs (Dict): Original keyword arguments for the executed task.
+
+        Returns:
+            None: The return value of this handler is ignored.
+        """
+
+    def on_retry(self, exc, task_id, args, kwargs, einfo):
+        """Retry handler.
+
+        This is run by the worker when the task is to be retried.
+
+        Arguments:
+            exc (Exception): The exception sent to :meth:`retry`.
+            task_id (str): Unique id of the retried task.
+            args (Tuple): Original arguments for the retried task.
+            kwargs (Dict): Original keyword arguments for the retried task.
+            einfo (~billiard.einfo.ExceptionInfo): Exception information.
+
+        Returns:
+            None: The return value of this handler is ignored.
+        """
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        """Error handler.
+
+        This is run by the worker when the task fails.
+
+        Arguments:
+            exc (Exception): The exception raised by the task.
+            task_id (str): Unique id of the failed task.
+            args (Tuple): Original arguments for the task that failed.
+            kwargs (Dict): Original keyword arguments for the task that failed.
+            einfo (~billiard.einfo.ExceptionInfo): Exception information.
+
+        Returns:
+            None: The return value of this handler is ignored.
+        """
```

Comment:
Does implementing these here as no-ops (as opposed to not implementing them at all) have any consequences we should be aware of?

Reply:
It shouldn't. They were copy/pasted from the base class in celery.